extension/history.js

```diff
@@ -20,107 +21,124 @@ export class Database {
     });
   }
 
   get(primaryKey, TBName) {
     return new Promise((resolve, reject) => {
-      const request = indexedDB.open(this.DBName);
+      const request = indexedDB.open(this.dbName);
       request.onsuccess = e => {
         const database = e.target.result;
         const transaction = database.transaction([TBName]);
         const objectStore = transaction.objectStore(TBName);
         const read = objectStore.get(primaryKey);
-
-        read.onerror = function(event) {
-          reject("Unable to retrieve data from database!");
-        };
-
-        read.onsuccess = function(event) {
-          log.info("get result:", read.result);
+        read.onsuccess = e => {
+          log.info("Got result:", read.result);
           resolve(read.result);
         };
+        read.onerror = e => {
+          const code = e.target.errorCode;
+          reject(new Error(`Unable to retrieve data from database: ${code}`));
+        };
         database.close();
       };
-      request.onerror = event => reject("Unable to retrieve from database!");
+      request.onerror = e => {
+        const code = e.target.errorCode;
+        reject(new Error(`Unable to retrieve from database: ${code}`));
+      };
     });
   }
 
   getAll(TBName) {
-    return new Promise(resolve => {
-      const request = indexedDB.open(this.DBName);
+    return new Promise((resolve, reject) => {
+      const request = indexedDB.open(this.dbName);
       request.onsuccess = e => {
         const database = e.target.result;
         const objectStore = database.transaction(TBName).objectStore(TBName);
         const list = [];
-
-        objectStore.openCursor().onsuccess = event => {
-          const cursor = event.target.result;
+        // return all objects from the object store according to `this.order`
+        const request = objectStore.openCursor(null, this.order);
+        request.onsuccess = e => {
+          const cursor = e.target.result;
           if (cursor) {
             list.push(cursor.value);
             cursor.continue();
           } else {
             resolve(list);
           }
         };
+        request.onerror = e => {
+          const code = e.target.errorCode;
+          reject(new Error(`Unable to retrieve data from database: ${code}`));
+        };
       };
     });
   }
 
   add(obj, TBName) {
     return new Promise((resolve, reject) => {
-      const request = indexedDB.open(this.DBName);
+      const request = indexedDB.open(this.dbName);
       request.onsuccess = e => {
         const database = e.target.result;
         const add = database
           .transaction([TBName], this.readwrite)
           .objectStore(TBName)
           .add(obj);
-        add.onsuccess = event => {
-          // log.info("1 record has been added to your database.");
-          resolve("1 record has been added to your database.");
+        add.onsuccess = e => {
+          log.info("1 record has been added to your database.");
+          resolve();
```

Comment:
Also here, should the user get back the entry they added? For example, if IndexedDB adds an `id` field, whoever calls add will not have `id` available to them.